go/protocols/materialize/lifecycle_test.go

```diff
@@ -0,0 +1,211 @@
+package materialize
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/bradleyjkemp/cupaloy"
+	"github.com/estuary/flow/go/protocols/fdb/tuple"
+	pf "github.com/estuary/flow/go/protocols/flow"
+	"github.com/stretchr/testify/require"
+	pb "go.gazette.dev/core/broker/protocol"
+	pc "go.gazette.dev/core/consumer/protocol"
+)
+
+//go:generate flowctl raw build --build-id test-build --db-path testdata/temp.db --source testdata/flow.yaml
+//go:generate sqlite3 file:testdata/temp.db "SELECT WRITEFILE('testdata/materialization.proto', spec) FROM built_materializations WHERE materialization = 'test/sqlite';"
+
+func TestStreamLifecycle(t *testing.T) {
```

Comment:
I'm not able to run these tests locally - seems that some of the structs like `Request_Open` are missing, and the function signature for some things is changed? For example, `WriteAcknowledged`.